### PR TITLE
Fix interpolation of lists and maps from resources

### LINKED ISSUE #7143: Unable to make splat reference to attribute of type map
### Terraform Version

master / 2127466280172a7cd0aa1d863cb932aa0107c986 

### Affected Resource(s)

core

### Terraform Configuration Files
```hcl
resource "aws_security_group" "foo" {
  count = 3
  tags {
    Name = "sg-${count.index}"
  }
}

resource "aws_instance" "web" {
  count                  = 2
  instance_type          = "t2.nano"
  ami                    = "ami-06b94666"
  vpc_security_group_ids = ["${aws_security_group.foo.*.id}"]
}

output "listoflists" { value = ["${aws_instance.web.*.vpc_security_group_ids}"] }
output "listofmaps" { value = ["${aws_security_group.foo.*.tags}"] }
```

### Steps to Reproduce

1. `TF_LOG=warn terraform apply`

### Expected Behavior

Outputs including lists of lists and list of maps.

### Actual Behavior

The `listoflists` output is flattened (perhaps this is intentional behavior?):

```
listoflists = [
  sg-553ce633
  sg-543ce632
  sg-523ce634
  sg-553ce633
  sg-543ce632
  sg-523ce634
]
```

And the `listofmaps` output cannot be referenced. Two warnings are printer:

```
2016/06/12 12:17:01 [WARN] Output interpolation "listofmaps" failed: Resource 'aws_security_group.foo' does not have attribute 'tags' for variable 'aws_security_group.foo.*.tags'
2016/06/12 12:17:01 [WARN] Output interpolation "listofmaps" failed: Resource 'aws_security_group.foo' does not have attribute 'tags' for variable 'aws_security_group.foo.*.tags'
```

### PR DESCRIPTION
**Pull request by jen20**
This PR introduces a fix for HIL and updates multi-variable interpolation from state to work correctly.

Fixes #7143, though there is a follow-up PR to fix output formatting for nested lists and maps.

### CODE REVIEW

**Comment by jen20** [comment]
cc @phinze

File: vendor/github.com/hashicorp/hil/convert.go [hunk 1]
```diff
@@ -86,3 +90,59 @@ func InterfaceToVariable(input interface{}) (ast.Variable, error) {
...
+				result[key] = convertedValue
+			} else {
+				return nil, err
+			}
+		}
+
+		return result, nil
+	}
+
+	return nil, fmt.Errorf("Find")
```
Comment (phinze) [inline vendor/github.com/hashicorp/hil/convert.go]: Should probably circle back and enhance this error a bit.

Comment (phinze) [inline vendor/github.com/hashicorp/hil/convert.go]: Not prereq to merging this PR tho.

**Comment by phinze** [comment]
LGTM

File: vendor/github.com/hashicorp/hil/convert.go (hunk 1 above)
Comment (jen20) [inline vendor/github.com/hashicorp/hil/convert.go]: Yup this is in HIL anyway, need to fix it there first.

**Pull request merged**